framework/Optimizers/gradients/CentralDifference.py

```diff
@@ -0,0 +1,68 @@
+import abc
+import copy
+import pandas as pd
+import numpy as np
+from utils import InputData, InputTypes, randomUtils, mathUtils
+
+from utils import InputData, InputTypes, randomUtils, mathUtils
+
+from .GradientApproximater import GradientApproximater
+
+from .FiniteDifference import FiniteDifference
+
+class CentralDifference(FiniteDifference):
+  
+  #i=0
+  def chooseEvaluationPoints(self, opt, stepSize):
+    """
+      Determines new point(s) needed to evaluate gradient
+      @ In, opt, dict, current opt point (normalized)
+      @ In, stepSize, float, distance from opt point to sample neighbors
+      @ Out, evalPoints, list(dict), list of points that need sampling
+      @ Out, evalInfo, list(dict), identifying information about points
+    """
+    dh = self._proximity * stepSize
+    evalPoints = []
+    evalInfo = []
+
+
+    for o, optVar in enumerate(self._optVars):
+ 
+      optValue = opt[optVar]
+      new = copy.deepcopy(opt)
+      delta = dh #* directions[o]
+
+      new[optVar] = optValue + delta
```

Comment:
You're only doing the positive have (optValue + delta); we need both positive and negative (optValue - delta) to be submitted as gradient evaluation points; otherwise, you won't get the points you need to calculate the gradient.

Reply:
Ok! This I removed after I heard back from you. Anyways I will remove it and revert back to my original setup.